### PR TITLE
[EasyActivity] Serialization fix

### PR DESCRIPTION
**Pull request by voodooism**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no 
| BC breaks?    | no     
| Deprecations? | no 
| Tests pass?   | yes

### CODE REVIEW

File: packages/EasyActivity/src/Interfaces/ActivitySubjectInterface.php [hunk 1]
```diff
@@ -11,12 +11,17 @@ public function getActivitySubjectId(): string;
...
      */
-    public function getAllowedActivityProperties(): ?array;
+    public function getAllowedActivityProperties(): array;
 
     /**
-     * @return array<string>|array<string, mixed>
+     * @return array<string, list<string>>
      */
-    public function getDisallowedActivityProperties(): ?array;
+    public function getNestedObjectAllowedProperties(): array;
```
Comment (roman-eonx) [inline packages/EasyActivity/src/Interfaces/ActivitySubjectInterface.php:21]: Let's make this method name consistent with the others: `getNestedObjectAllowedActivityProperties`.

File: packages/EasyActivity/src/Bridge/Symfony/Serializers/SymfonyActivitySubjectDataSerializer.php [hunk 2]
```diff
@@ -48,7 +50,7 @@ public function serialize(array $data, ActivitySubjectInterface $subject): ?stri
         $context = [];
 
         foreach ($data as $key => $value) {
-            if ($allowedProperties !== null
+            if ($allowedProperties !== []
```
Comment (roman-eonx) [inline packages/EasyActivity/src/Bridge/Symfony/Serializers/SymfonyActivitySubjectDataSerializer.php:53]: This change makes me think we shouldn't make the method return value non-nullable. What if someone wants to explicitly set the list of allowed properties to an empty one (meaning "log nothing")? `allowedProperties === null` seems more explicit to me, as it is read as "allowed properties are not defined".

**Review by roman-eonx — CHANGES_REQUESTED** [review]

File: packages/EasyActivity/src/Interfaces/ActivitySubjectInterface.php (hunk 1 above)
Comment (voodooism) [reply packages/EasyActivity/src/Interfaces/ActivitySubjectInterface.php]: done

File: packages/EasyActivity/src/Bridge/Symfony/Serializers/SymfonyActivitySubjectDataSerializer.php (hunk 2 above)
Comment (voodooism) [reply packages/EasyActivity/src/Bridge/Symfony/Serializers/SymfonyActivitySubjectDataSerializer.php]: done

**Review by mvlad — DISMISSED** [review]

**Review by akl-95 — CHANGES_REQUESTED** [review]

File: packages/EasyActivity/src/Interfaces/ActivitySubjectInterface.php [hunk 3]
```diff
@@ -11,12 +11,17 @@ public function getActivitySubjectId(): string;
...
+     * @return list<string>|array<string, list<string>>|null
      */
     public function getAllowedActivityProperties(): ?array;
 
     /**
-     * @return array<string>|array<string, mixed>
+     * @return array<string, list<string>>
      */
-    public function getDisallowedActivityProperties(): ?array;
+    public function getNestedObjectAllowedActivityProperties(): array;
```
[the remark this answers is not in the document]
Comment (voodooism) [reply packages/EasyActivity/src/Interfaces/ActivitySubjectInterface.php]: done

File: packages/EasyActivity/src/Bridge/BridgeConstantsInterface.php [hunk 4]
```diff
@@ -25,4 +25,9 @@ interface BridgeConstantsInterface
      * @var string
      */
     public const PARAM_TABLE_NAME = 'easy_activity.table_name';
+
+    /**
+     * @var string
+     */
+    public const SERVICE_ALIAS_SERIALIZER = 'easy_activity.serializer';
```
[the remark this answers is not in the document]
Comment (voodooism) [reply packages/EasyActivity/src/Bridge/BridgeConstantsInterface.php]: done

**Review by akl-95 — DISMISSED** [review]

**Review by roman-eonx — DISMISSED** [review]

File: packages/EasyActivity/src/Bridge/Symfony/Serializers/SymfonyActivitySubjectDataSerializer.php [hunk 5]
```diff
@@ -48,7 +55,7 @@ public function serialize(array $data, ActivitySubjectInterface $subject): ?stri
         $context = [];
 
         foreach ($data as $key => $value) {
-            if ($allowedProperties !== null
+            if (empty($allowedProperties) === false
```
Comment (itorgov) [inline packages/EasyActivity/src/Bridge/Symfony/Serializers/SymfonyActivitySubjectDataSerializer.php:58]: Please use `\count` instead.

**Review by itorgov — CHANGES_REQUESTED** [review]

File: packages/EasyActivity/src/Bridge/Symfony/Serializers/SymfonyActivitySubjectDataSerializer.php (hunk 5 above)
Comment (voodooism) [reply packages/EasyActivity/src/Bridge/Symfony/Serializers/SymfonyActivitySubjectDataSerializer.php]: done

Comment (voodooism) [reply packages/EasyActivity/src/Bridge/Symfony/Serializers/SymfonyActivitySubjectDataSerializer.php]: done

**Review by natepage — APPROVED** [review]

**Pull request merged**